applications/nrf_desktop/src/hw_interface/wheel.c

```diff
@@ -73,6 +63,39 @@ static enum state state;
 static int enable_qdec(enum state next_state);
 
 
+static const struct device *map_gpio_port(int pin_number)
+{
+	__ASSERT_NO_MSG(pin_number >= 0);
+
+	size_t port_idx = pin_number / PINS_PER_GPIO_PORT;
+	const struct device *dev = NULL;
+
+	switch (port_idx) {
+	case 0:
+		dev = DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio0));
+		break;
+
+	case 1:
+		dev = DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio1));
+		break;
+
+	default:
+		break;
+	}
+
+	/* Assert proper configuration. */
+	__ASSERT_NO_MSG(dev);
+
+	return dev;
+}
+
+static gpio_pin_t map_gpio_pin(int pin_number)
+{
+	__ASSERT_NO_MSG(pin_number >= 0);
+
+	return pin_number % PINS_PER_GPIO_PORT;
+}
```

Comment:
Could we use the following NRF_PSEL encoding to fetch the information that encodes the QDEC pins?

https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/dt-bindings/pinctrl/nrf-pinctrl.h

```
/*
 * The whole nRF pin configuration information is encoded in a 32-bit bitfield
 * organized as follows:
 *
 * - 31..16: Pin function.
 * - 15:     Reserved.
 * - 14:     Pin inversion mode.
 * - 13:     Pin low power mode.
 * - 12..9:  Pin output drive configuration.
 * - 8..7:   Pin pull configuration.
 * - 6..0:   Pin number (combination of port and pin).
 */
```

Then, we could build the switch case to support arbitrary number of GPIO ports

Reply:
These mapping functions are also used for `enable_pin` from `nrf-qdec`: https://docs.zephyrproject.org/latest/build/dts/api/bindings/sensor/nordic,nrf-qdec.html

The pin definition does not follow PSEL's approach and I wanted to avoid creating a separate mapping function. Can we keep it as is?

Reply:
Discussed offline. Agreed to keep solution as is for now. We could work on further improvements later under: `NCSDK-26235`